pkg/tagger/collectors/garden_extract.go

```diff
@@ -0,0 +1,30 @@
+package collectors
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func (c *GardenCollector) extractTags() (tagsByInstanceGUID map[string][]string, err error) {
+	if c.clusterAgentEnabled {
+		tagsByInstanceGUID, err = c.dcaClient.GetAllCFAppsMetadata()
+		if err != nil {
+			return tagsByInstanceGUID, err
+		}
+	} else {
+		log.Debugf("Cluster agent not enabled or misconfigured, tagging CF app with container id only")
+		gardenContainers, err := c.gardenUtil.GetGardenContainers()
+		if err != nil {
+			return tagsByInstanceGUID, fmt.Errorf("cannot get container list from local garden API: %v", err)
+		}
+		tagsByInstanceGUID = make(map[string][]string, len(gardenContainers))
+		for _, gardenContainer := range gardenContainers {
+			tagsByInstanceGUID[gardenContainer.Handle()] = []string{
+				fmt.Sprintf("container_name:%s", gardenContainer.Handle()),
+				fmt.Sprintf("app_instance_guid:%s", gardenContainer.Handle()),
```

Comment:
not familiar with cloud foundry, but is it intended to have these two tags with the same value `gardenContainer.Handle()` ?

Reply:
This `container_name` tag value is a fallback for when the DCA agent isn't enabled, just so that we don't have an empty container name (empty container name can cause issues in the UI). 
Let me know if that's not a proper option

Reply:
sounds good, thanks for the clarification - is there a way to get the container name from the local api? wouldn't expect to see a GUID as container name in the UI

Reply:
AFAICS, there is no way to have the metadata from the local API, we have to query the BBS API to get it. Since it's not something we want to do for each node agent for performance reasons, this is the fallback that I saw.
However, we'll make it so that using our tile to deploy the agent in WMWare CloudFoundry will automatically deploy an instance of the DCA whenever process collection for containers is enabled, and we'll document that it's needed to have the best experience when manually deploying agents using open source CloudFoundry.
This is mostly a safeguard in case the cluster agent isn't enable. I thought it would be better than a hard fail and no container collection at all